R/readepi_helpers.R

```diff
@@ -130,3 +128,40 @@ fingertips_get_args <- function(args_list =
     parent_area_type_id = parent_area_type_id
   )
 }
+
+#' Check if the base_url argument is specified as a correct URL
+#'
+#' @param base_url the URL to the HIS of interest
+#'
+#' @keywords internal
+#'
+url_check <- function(base_url) {
+  checkmate::assert_character(base_url, any.missing = FALSE, len = 1L,
+                              null.ok = FALSE)
+  if (startsWith(base_url, "https://")) {
+    base_url   <- gsub("https://", "", base_url, fixed = TRUE)
+  }
+
+  # split the URL parts
+  url_parts    <- unlist(strsplit(base_url, "/", fixed = TRUE))
+  # without the protocol, lets look for the sub-domain, second-level domain
+  # and the top-level domain
+  domain_structure_is_correct <- url_check_domain_structure(url_parts[[1L]])
+  stopifnot("Incorrect domain name in provided URL!" =
+              domain_structure_is_correct)
+}
+
+#' Check whether the domain structure of the URL is correct
+#'
+#' @param url_parts a character string with the domain from the URL
+#'
+#' @keywords internal
+url_check_domain_structure <- function(url_parts) {
+  domain_structure_is_correct   <- TRUE
+  url_domains                   <- unlist(strsplit(url_parts, ".",
+                                                   fixed = TRUE))
+  if (length(url_domains) < 3L) {
+    domain_structure_is_correct <- FALSE
+  }
```

Comment:
I don't follow. If the database is hosted at `example.com`, or `https://example.com` if we use the URL, this will not work.

There is no guarantee that database will be hosted on a subdomain.

Reply:
I will reduce this to `2` to avoid issues in the future.